cmd/gophermart/main.go

```diff
@@ -1,3 +1,74 @@
 package main
 
-func main() {}
+import (
+	"context"
+	"net/http"
+	"os"
+	"os/signal"
+	"syscall"
+
+	"github.com/SversusN/gophermart/config"
+	agent "github.com/SversusN/gophermart/internal/accrualagent/service"
+	app "github.com/SversusN/gophermart/internal/app"
+	handler "github.com/SversusN/gophermart/internal/controller/http/handlers"
+	repository "github.com/SversusN/gophermart/internal/repository"
+	psql "github.com/SversusN/gophermart/internal/repository/psql"
+	"github.com/SversusN/gophermart/internal/service"
+	"github.com/SversusN/gophermart/pkg/logger"
+)
+
+func main() {
+	log, err := logger.InitLogger()
+	if err != nil {
+		log.Fatal("error init logger")
+	}
+
+	defer log.Sync()
+	zp := log.Sugar()
+
+	conf, err := config.NewConfig()
+	if err != nil {
+		zp.Fatalf("failed to retrieve env variables, %v", err)
+	}
+
+	db, err := psql.NewPsql(conf.DatabaseURI)
+	if err != nil {
+		zp.Fatalf("DB connection error %v", err)
+	}
+
+	err = db.Init(conf.DatabaseURI)
+	if err != nil {
+		zp.Fatalf("failed to create db table %v", err)
+	}
+
+	ctx, stopping := context.WithCancel(context.Background())
+	defer stopping()
+
+	repos := repository.NewRepository(db.DB, log)
+	services := service.NewService(repos, log)
+	handlers := handler.NewHandler(services, log)
+	//настройка воркера
+	agentRepo := repository.NewAgentRepository(db.DB, log)
+	newAgent := agent.NewAgent(agentRepo, conf.AccrualSystemAddress, log)
+	newAgent.Start(ctx)
```

Comment:
```suggestion
	wg := sync.WaitGroup
	newAgent.Start(ctx, wg)
```
В агенте поставить defer wg.Done(), а в процедуре шатдауна - wg.Wait(), чтобы нормально дождаться завершения работы агента.

Reply:
+